src/internals/tuples/Tuples.ts

```diff
@@ -260,6 +255,26 @@ export namespace Tuples {
     return: ReduceImpl<Extract<this["arg0"], Fn>, this["arg1"], this["arg2"]>;
   }
 
+  type ReverseImpl<tuple> = tuple extends [infer first, ...infer rest]
+    ? [...ReverseImpl<rest>, first]
```

Comment:
This is not tail recursive optimised. Add an Acc and pass accumulated array to it.

Reply:
thanks for the feedback! I updated it.